src/Shared/InstrumentationScopeHelper.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation;
+
+internal static class InstrumentationScopeHelper
```

Comment:
@Kielek We have similar logic here:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f4c73adb1f9845eb8e9553d7d21c5a2c629413f/src/OpenTelemetry/Sdk.cs#L34-L35

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f4c73adb1f9845eb8e9553d7d21c5a2c629413f/src/OpenTelemetry/Sdk.cs#L114

Also here but it seems a little different:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f4c73adb1f9845eb8e9553d7d21c5a2c629413f/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs#L232

This new helper class doesn't _really_ have anything to do with instrumentation scopes. Could we perhaps rename it something like AssemblyVersionHelper and refactor the SDK version stuff (maybe OtlpExporter stuff too) so that everything is using the same code for versioning?

Doesn't have to be on this PR.

The API could also be an extension method. Something like...

```csharp
internal static class AssemblyVersionExtensions
{
    public static string GetPackageVersion(this Assembly assembly)
    {
        // not shown
    }
}
```

In the current signature you pass a type to use to lookup the assembly. I don't think that is immediately obvious IMO passing the assembly to use is nicer/more explicit.

Reply:
Most of the changes covered in 9f5f5bfbee10d81366eb92d3e6dfceb175b68b3b and d8f9a1a11379097b480cc3c0d20a1388dbb808a1.

I think that changes in OTLP exported, related to dropping hash commit, deservers CHANGELOG entry.
It will bring as a bit closer to recommendation, to put there just version, without any additional data such as commit hash.
I will create follow up PR, when this one will be merged.

Reply:
User agent. Now it contains haskcommit.

Before changes OTel-OTLP-Exporter-Dotnet/version+hashcommit
After-changes  OTel-OTLP-Exporter-Dotnet/version